### ISSUE TITLE
VariantSet metadata not correctly filled in

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#677]
If you run
```
$ python client_dev.py variantsets-search http://localhost:8000 -O json | jq . | less
```
you can see a bunch of VariantSetMetadata (see [variants.avdl](https://github.com/ga4gh/schemas/blob/master/src/main/resources/avro/variants.avdl) for details) fields that look like this
```json
{
      "info": {},
      "description": "Confidence interval around END for imprecise variants",
      "number": "2",
      "value": "",
      "key": "INFO.CIEND",
      "type": "Integer",
      "id": ""
  },
```
The ID field here should be a combination of the variantset ID and the key, and use the usual obfuscation patterns. Object IDs in the server spell out the object hierarchy above a give object, and are obfuscated to prevent clients from mistakenly thinking this was a requirement of the protocol and making use of this structure.

- Open ``ga4gh/datamodel/__init__.py`` and add a new class ``VariantSetMetadataCompoundId`` that inherits from VariantSetCompoundId. The extra field should be "key". (Follow the definition of CallSetCompundId).

- Open ``ga4g/datamodel/variants.py`` and find the _getMetadataFromVcf function. Add the definition of id using the VariantSetMetadataCompoundId. (Base this definition on the code in ga4gh/datamodel/reads.py for getExperimentId().

- Make tests  for this in tests/datadriven/test_variants.py in the testVariantSetMetadata method. It may be helpful to put in ``print`` statements to see what's going on. Run the tests using ``nosetests -vs tests/datadriven/test_variants.py``. (It's not possible to further subset these tests, unfortunately, because they depend on custom code to apply each test to each data file.) The test should look something like

```python
deobfuscatedId = datamodel.CompoundId.deobfuscate(metadata.id))
self.assertTrue(deobfuscatedId.endswith, key)
```

- Add a test in tests/unit/test_compound_ids.py to check the VariantSetMetadataCompoundId. Model the test on ``testExperimentParse``.

**Comment by pcingola** [comment]
Working on this.

**Comment by jeromekelleher** [comment]
Fixed by #733.
